COMETwebapp/ViewModels/Components/Common/OpenTab/OpenTabViewModel.cs

```diff
@@ -103,10 +104,11 @@ public TabbedApplication SelectedApplication
         /// <summary>
         /// Opens the <see cref="EngineeringModel" /> based on the selected field
         /// </summary>
-        /// <returns>A <see cref="Task" /> containing the operation <see cref="Result" /></returns>
-        public override async Task<Result<Iteration>> OpenSession()
+        /// <returns>A <see cref="Task" /></returns>
+        public async Task OpenTab()
         {
             var result = new Result<Iteration>();
+            var isIteration = this.SelectedApplication?.ViewModelType.IsAssignableTo(typeof(ISingleIterationApplicationBaseViewModel)) == true;
```

Comment:
Could you check based on the fact that the ObjectOfInterest is Iteration? Would be more performant

Reply:
Sure! Done in the latest commit :)